lib/plugins/menu/custom-events.js

```diff
@@ -0,0 +1,21 @@
+export const dispatchSelectCustomEvent = (select) => {
+  select.dispatchEvent(
+    new CustomEvent("visualEditorSelectChange", {
+      bubbles: true,
+      detail: {
+        selectText: select.options[select.selectedIndex].text,
```

Comment:
Would this make more sense to be "selected option"? I also feel a little weird that we're passing the `text` around and not the `value`, which is what I would want to match on if I was the consumer (I have a strong aversion to doing unformatted string comparisons). Then I would do a mapping from `value` to the text I wanted on the consumer side. What do you think? Yes it's an extra step on the Whitehall side, but it decouples the visual editor from Whitehall